man/fi_getinfo.3.md

```diff
@@ -338,6 +338,11 @@ additional optimizations.
   allow an initiator to target (or name) a specific receive context as
   part of a data transfer operation.
 
+*FI_PEER*
```

Comment:
Are there changes which use this or needed updates to fabtests or the common flow?

Reply:
What do you mean?

Reply:
This updates the man page only.  There's no code updated anywhere to use the new definition.

Reply:
Ah gotcha. Yes this is all targeted for the link provider as a way to query providers that it is able to link together. But other providers can use it as well if using a peer provider directly (like efa should TECHNICALLY add FI_PEER to its shm fi_getinfo call). I can add that.

Reply:
It's okay if the link provider is going to set this on input.  But what provider is going to respond to the query?  How does setting this not just fail the call?

Reply:
I don't follow your question - the link provider needs to make sure that all providers it's going to link together support the peer API so it queries the provider with caps->FI_PEER. shm will succeed. cxi will succeed. rxm will succeed. But for example psm3 probably won't. So the fi_getinfo call in the link layer will not include psm3 as a possible link. It will only include rxm+shm, cxi+shm, etc. If a user requests link with psm3+shm, it would return ENODATA instead of attempting to run it and getting an ENOSYS error. Are you saying you prefer link to fail when trying to open everything instead of not including the bad combination in the list of possible links? Or are you saying I missed adding this in shm (as a supported capability) and efa (as a requested shm capability)?

Reply:
E.g, in the shm provider:
```
#define SMR_TX_CAPS (OFI_TX_MSG_CAPS | FI_TAGGED | OFI_TX_RMA_CAPS | FI_ATOMICS)
#define SMR_RX_CAPS (FI_SOURCE | FI_RMA_EVENT | OFI_RX_MSG_CAPS | FI_TAGGED | \
		     OFI_RX_RMA_CAPS | FI_ATOMICS | FI_DIRECTED_RECV | \
		     FI_MULTI_RECV)

struct fi_info smr_info = {
	.caps = SMR_TX_CAPS | SMR_RX_CAPS | FI_MULTI_RECV | FI_LOCAL_COMM,
```

Where is FI_PEER being used as a cap anywhere?  I don't see any providers setting it.  It's not part of any common checks as a primary cap (e.g. included in ofi.h).  Fabtests getinfo_test doesn't include this.

Reply:
Ok so option number 2, ie you're grumpy I didn't add it. Got it. Will add.